### PR TITLE
adding service account support

### PR DESCRIPTION
**Pull request by lcouzens**
This adds support for C.R.C service account authentication to all filter scripts/docs.

### CODE REVIEW

File: docs/gcp/scripts/gcp-function-non-partition-dates.txt [hunk 1]
```diff
@@ -26,13 +26,13 @@ day = now.strftime("%d")
...
 
 # Required vars to update
-USER = os.getenv('username')          # Cost management username
-PASS = os.getenv('password')          # Cost management password
-SOURCE_ID = "CHANGE-ME"               # Cost management source_id
-BUCKET = "CHANGE-ME"                  # Filtered data GCP Bucket
-PROJECT_ID = "CHANGE-ME"              # Your project ID
-DATASET = "CHANGE-ME"                 # Your dataset name
-TABLE_ID = "CHANGE-ME"                # Your table ID
+CLIENT_ID = os.getenv('client_id')          # C.R.C service account client_id`
```
Comment (lcouzens) [inline docs/gcp/scripts/gcp-function-non-partition-dates.txt]:
```suggestion
CLIENT_ID = os.getenv('client_id')          # C.R.C service account client_id
```

File: docs/gcp/scripts/gcp-function.txt [hunk 2]
```diff
@@ -27,13 +27,13 @@ day = now.strftime("%d")
...
 
 # Required vars to update
-USER = os.getenv('username')          # Cost management username
-PASS = os.getenv('password')          # Cost management password
-SOURCE_ID = "CHANGE-ME"               # Cost management source_id
-BUCKET = "CHANGE-ME"                  # Filtered data GCP Bucket
-PROJECT_ID = "CHANGE-ME"              # Your project ID
-DATASET = "CHANGE-ME"                 # Your dataset name
-TABLE_ID = "CHANGE-ME"                # Your table ID
+CLIENT_ID = os.getenv('client_id')          # C.R.C service account client_id`
```
Comment (lcouzens) [inline docs/gcp/scripts/gcp-function.txt]:
```suggestion
CLIENT_ID = os.getenv('client_id')          # C.R.C service account client_id
```

File: docs/azure/azure.rst [hunk 3]
```diff
@@ -105,7 +105,7 @@ Key vault Credentials
 7. Review and create Vault
 8. Click on new vault and select secrets in the blade
 9. Generate/import two new secrets
-    i. Secret names as follows *UsernameFromVault* and *PasswordFromVault*
+    i. Secret names as follows *ClientIdFromVault* and *ClientSecretFromVault*
     ii. Giving them a secret value matching your console.redhat.com username and password respectively
```
Comment (djnakabaale) [inline docs/azure/azure.rst]: Should this be `Giving them a secret value matching your console.redhat.com client_id and client_secret respectively`?

**Review by djnakabaale — COMMENTED** [review]
A few comments, otherwise looks good to me:

We probably should add a note that the service account needs to have the appropriate permissions.

File: docs/azure/azure.rst (hunk 3 above)
Comment (lcouzens) [reply docs/azure/azure.rst]: This is something the user creates in Azure, effectively just variable names.

**Comment by lcouzens** [comment]
> A few comments, otherwise looks good to me:
> 
> We probably should add a note that the service account needs to have the appropriate permissions.

Right I wondered that, to be fair the docs here are to aid the official docs, and so far we don't specify that over there. Though we probably should :+1:

File: docs/aws/aws.rst [hunk 4]
```diff
@@ -2,6 +2,10 @@
 ===========================
 AWS: Customer-filtered data
 ===========================
+*Prerequisites:*
+    - A console.redhat.com service account is required for this Setup
```
Comment (lcouzens) [inline docs/aws/aws.rst]:
```suggestion
    - A console.redhat.com service account is required
```

File: docs/azure/azure.rst [hunk 5]
```diff
@@ -6,6 +6,9 @@ Azure: Customer-filtered data
 
 Azure resource group and storage account
 ========================================
+*Prerequisites:*
+    - A console.redhat.com service account is required for this Setup
```
Comment (lcouzens) [inline docs/azure/azure.rst]:
```suggestion
    - A console.redhat.com service account is required
```

File: docs/gcp/gcp.rst [hunk 6]
```diff
@@ -2,6 +2,9 @@
 ===========================
 GCP: Customer-filtered data
 ===========================
+*Prerequisites:*
+    - A console.redhat.com service account is required for this Setup
```
Comment (lcouzens) [inline docs/gcp/gcp.rst]:
```suggestion
    - A console.redhat.com service account is required
```

**Pull request merged**